client/src/research/EmailLinkLoginPage.js

```diff
@@ -0,0 +1,124 @@
+import React, { Component } from 'react';
+import './LoginPage.css';
+import Interactions from './Interactions.js';
+
+
+// This is the landing page users reach when clicking on a login 
+// link from their email. Users can confirm their email to get 
+// access to participant data.
+class EmailLinkLoginPage extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      email: "",
+      token: "default",
+      status: "default",
+      message: "Welcome back to Swipe Right!"
+    };
+
+    this.onUpdateEmail = this.onUpdateEmail.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+    this.getQueryVariable = this.getQueryVariable.bind(this);
+    this.authenticate = this.authenticate.bind(this);
+  }
+
+  getQueryVariable(variable) {
+    const params = {};
+    const rawParam = window.location.search;
+    if (rawParam) {
```

Comment:
Sorry if I forgot to mention this before - this parsing code is good but let's use the `querystring` library to parse this for us (here's an example: https://github.com/mit-teaching-systems-lab/swipe-right-for-cs/blob/012d4aba04bdfa9e3435d85552f703bed25e30ff/client/src/GamePage.js#L39).

Reply:
done!